### PR TITLE
Legalize fcvt_from_u/sint.f32/f64.i8/i16

### PR DESCRIPTION
**Pull request by bjorn3**
cc #466 (seems like I didn't list the instructions there though)

### CODE REVIEW

File: filetests/isa/x86/legalize-fcvt_from_usint-i16.clif [hunk 1]
```diff
@@ -0,0 +1,14 @@
+test compile
+target x86_64
+
+function u0:0(i16) -> f64 fast {
+ebb0(v0: i16):
+    v1 = fcvt_from_uint.f64 v0
```
Comment (bnjbvr) [inline filetests/isa/x86/legalize-fcvt_from_usint-i16.clif]: Can you add check statements for what's expected in place of this instruction?

File: filetests/isa/x86/legalize-fcvt_from_usint-i16.clif [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    return v1
+}
+
+function u0:1(i16) -> f64 fast {
+ebb0(v0: i16):
+    v1 = fcvt_from_sint.f64 v0
```
Comment (bnjbvr) [inline filetests/isa/x86/legalize-fcvt_from_usint-i16.clif]: Ditto here.

File: cranelift-codegen/meta-python/base/legalize.py [hunk 3]
```diff
@@ -261,6 +261,23 @@ def widen_imm(signed, op):
 for binop in [sdiv_imm, srem_imm]:
     widen_imm(True, binop)
 
+# Use expand instead of widen, because widen only gets applied for i8 and i16
+# base type vars, but these take f32 and f64 as base type var.
+for ty in [types.f32, types.f64]:
```
Comment (bnjbvr) [inline cranelift-codegen/meta-python/base/legalize.py]: Can you move this chunk of code next to the other expand legalizations, please?

**Comment by bnjbvr** [comment]
Hi @bjorn3 ! Now that the legalization port to Rust is over, you should be able to define it in Rust (and still need to define it in Python for the encoding table, I'm afraid). Would you like to take care of this, or do you want me to take over?

**Comment by bjorn3** [comment]
I am quite busy this and next week, so please take over.

**Comment by bnjbvr** [comment]
Closing in favor of #768; the filetest checks couldn't be added because there's actually a lot of code that's added to expand the fcvt_from_sint/uint. To wit: these instructions have the wasm semantics if I remember correctly, so you want to make sure that they work as you expect for edge case values.

**Pull request closed without merging**